### PR TITLE
Create Login Modal

### PR DESCRIPTION
**Pull request by supthefuhrer1**
Added Login Modal

### CODE REVIEW

File: src/app/page.tsx [hunk 1]
```diff
@@ -1,10 +1,19 @@
...
+import { Button } from "@mantine/core";
+import { useState } from "react";
+import LoginComponent from "@/components/Login";
 
 export default function Home() {
+  const [open, setOpen] = useState(false)
   return (
     <main>
       <h1>Home</h1>
+      <Button onClick={() => setOpen(prev => !prev)}>
```
Comment (shelly19gkmit) [inline src/app/page.tsx:13]: create a function to handle setOpen(prev => !prev)

File: src/app/page.tsx [hunk 2]
```diff
@@ -1,10 +1,19 @@
...
+        Login
+      </Button>
+      <LoginComponent isIndianUser={true} open={open} onClose={() => setOpen(false)} />
```
Comment (shelly19gkmit) [inline src/app/page.tsx:16]: rename to isOpen

File: src/components/CountryPicker/index.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            wrapper: styles['picker-wrapper'],
+            input: styles['picker-input'],
+            dropdown: styles['picker-dropdown'],
+        }}
+      value={selectedCountry}
+      onChange={(value, option) => onChangeHandler(value)}
+      data={countryCodes.map((item, index) => ({
+        label: `${item.flag} ${item.dial_code}`,
+        value: item.code,
+      }))}
```
Comment (shelly19gkmit) [inline src/components/CountryPicker/index.tsx:30]: create a separate function and use it here

File: src/components/Login/index.tsx [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        align="center"
+        justify="space-between"
+      >
+        <Box w={342}>
+          <Flex mt={36} mb={40} justify={"center"}>
+            <Text c={"white"} size="32px">
+              Login
+            </Text>
+          </Flex>
+          {isIndianUser ? (
```
Comment (shelly19gkmit) [inline src/components/Login/index.tsx:72]: create a function for this and render it here

File: src/components/Login/index.tsx [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+              Terms and Conditions
+            </span>{" "}
+          </Text>
+        </Flex>
+      </Flex>
+    </Drawer>
+  );
+};
+
+const NumberInput: React.FC<NumberInputProps> = ({
```
Comment (shelly19gkmit) [inline src/components/Login/index.tsx:141]: maintain separate file for NumberInput, rename to PhoneNumberInput

File: src/components/SocialConnectButton/index.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    <Button
+      onClick={onClick}
+      classNames={{
+        root: styles["social-button-root"],
+      }}
+      styles={{
+        root: {
+          width,
+        },
+      }}
```
Comment (shelly19gkmit) [inline src/components/SocialConnectButton/index.tsx:29]: styles should be maintained in separate file, don't add at component level

File: yarn.lock
Comment (shelly19gkmit) [inline yarn.lock:1]: remove yarn.lock

File: src/components/Login/index.tsx [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+} from "@mantine/core";
+import styles from "./Login.module.css";
+import closeIcon from "../../assets/close.svg";
+import Image from "next/image";
+import CountryPicker from "../CountryPicker";
+import twitterLogo from "../../assets/twitter-original logo.svg";
+import googleLogo from "../../assets/Google.svg";
+import whatsappLogo from "../../assets/whatsapp.svg";
+import discordLogo from "../../assets/Discord logo.svg";
+import SocialConnectButton from "../SocialConnectButton";
```
Comment (shelly19gkmit) [inline src/components/Login/index.tsx:20]: plz create alias to maintain code quality

File: src/components/CountryPicker/CountryPicker.module.css [hunk 8]
```diff
@@ -0,0 +1,15 @@
+.picker-wrapper {
```
Comment (mandeepatindigg) [inline src/components/CountryPicker/CountryPicker.module.css:1]: file name to camelCase

File: src/components/Login/Login.module.css [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+    border: none;
+    color: #fff;
+  }
+}
+.input-wrapper {
+  width: 100%;
+}
+
+.otp-input-root {
+  border: 1px solid #415f66;
```
Comment (mandeepatindigg) [inline src/components/Login/Login.module.css:41]: use colors from css color variables

File: package-lock.json
Comment (shelly19gkmit) [inline package-lock.json:1]: Remove this file from PR

File: src/components/Login/index.tsx [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+          <Image alt="close" width={32} height={32} src={closeIcon} />
+        </ActionIcon>
+      </Flex>
+      <Flex
+        h={'90vh'}
+        w={'100%'}
+        direction={'column'}
+        align="center"
+        justify="space-between">
+        {showOTPScreen ? (
```
Comment (shelly19gkmit) [inline src/components/Login/index.tsx:74]: create a function and call here to render component

File: src/components/Login/index.tsx [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+          <VerifyOTP selectedCountry={countryCode} phoneNumber={mobileNo} />
+        ) : (
+          <>
+            <Box w={342}>
+              <Flex mt={36} mb={40} justify={'center'}>
+                <Text c={'white'} size="32px">
+                  Login
+                </Text>
+              </Flex>
+              {isIndianUser ? (
```
Comment (shelly19gkmit) [inline src/components/Login/index.tsx:84]: Don't use ternary operator inside return statement, create functions to handle these conditions

**Pull request merged**